ui/tests/integration/components/kv/page/kv-page-version-diff-test.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { setupEngine } from 'ember-engines/test-support';
+import { setupMirage } from 'ember-cli-mirage/test-support';
+import { click, findAll, render } from '@ember/test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
+import { kvMetadataPath, kvDataPath } from 'vault/utils/kv-path';
+import { PAGE } from 'vault/tests/helpers/kv/kv-selectors';
+import { allowAllCapabilitiesStub } from 'vault/tests/helpers/stubs';
+
+module('Integration | Component | kv | Page::Secret::Metadata::VersionDiff', function (hooks) {
+  setupRenderingTest(hooks);
+  setupEngine(hooks, 'kv');
+  setupMirage(hooks);
+
+  hooks.beforeEach(async function () {
+    this.store = this.owner.lookup('service:store');
+    this.server.post('/sys/capabilities-self', allowAllCapabilitiesStub());
+    this.server.get(`/kv-engine/data/my-secret?version=1`, () => {
+      return {
+        request_id: 'foobar',
+        data: {
+          data: { hello: 'world' },
+          metadata: {
+            created_time: '2023-06-20T21:26:47.592306Z',
+            custom_metadata: null,
+            deletion_time: '',
+            destroyed: false,
+            version: 1,
+          },
+        },
+      };
+    });
+
+    const metadata = this.server.create('kv-metadatum');
+    metadata.id = kvMetadataPath(this.backend, this.secret);
+    this.store.pushPayload('kv/metadata', {
+      modelName: 'kv/metadata',
+      ...metadata,
+    });
+
+    this.backend = 'kv-engine';
+    this.secret = 'my-secret';
+    this.metadata = this.store.peekRecord('kv/metadata', metadata.id);
+    this.breadcrumbs = [{ label: 'version history', route: 'secret.metadata.versions' }, { label: 'diff' }];
+
+    // compare version 4
+    const dataId = kvDataPath(this.backend, this.secret, 4);
+    this.store.pushPayload('kv/data', {
+      modelName: 'kv/data',
+      id: dataId,
+      secret_data: { foo: 'bar' },
+      created_time: '2023-07-20T02:12:17.379762Z',
+      custom_metadata: null,
+      deletion_time: '',
+      destroyed: false,
+      version: 4,
+    });
+  });
+
+  test('it renders compared data of the two versions and shows icons for deleted, destroyed and current', async function (assert) {
+    assert.expect(12);
+
+    await render(
+      hbs`
+       <Page::Secret::Metadata::VersionDiff
+        @metadata={{this.metadata}} 
+        @path={{this.secret}}
+        @backend={{this.backend}}
+        @breadcrumbs={{this.breadcrumbs}}
+      />
+      `,
+      { owner: this.engine }
+    );
+    const [left, right] = findAll(PAGE.detail.versionDropdown);
+    assert.dom(PAGE.diff.visualDiff).hasText(
+      `{ \"foo\": \"bar\" }`, // eslint-disable-line no-useless-escape
+      'correctly pull in the data from version 4 and compared to version 1.'
+    );
+    assert.dom(PAGE.diff.deleted).hasText(`foo"bar"`);
```

Comment:
case in point: I'd expect this to say `hello"world"` since the newest version is foo: bar

Reply:
yeah - I think this makes a lot of sense!